wire-ios/Wire-iOS/Sources/UserInterface/ContactsUI/ContactsViewController.swift

```diff
@@ -205,12 +204,20 @@ final class ContactsViewController: UIViewController {
         let willAppear = (beginFrame.minY - endFrame.minY) > 0
         let padding: CGFloat = 12
 
-        UIView.animate(withKeyboardNotification: notification, in: view, animations: { [weak self] keyboardFrame in
+        // Calculate the effective keyboard height
+        var keyboardHeight = view.frame.intersection(endFrame).height
+
+        // On iPad, add an estimated accessory height if applicable
+        if traitCollection.userInterfaceIdiom == .pad {
+            keyboardHeight += 44  // Approximate height of an input accessory view
```

Comment:
good point, but I haven't found any way to get it so far

Reply:
I guess what I expect is that the `keyboardWillChangeFrameNotification` would include the height of any accessory. Did you see that this is no happening in practice?

Otherwise have you considered:

- Inheriting from `UITableViewController` - I think this takes care of keyboard avoidance completely.
- Using [UIKeyboardLayoutGuide](https://developer.apple.com/documentation/uikit/uikeyboardlayoutguide) instead of responding to notifications.

Reply:
used [UIKeyboardLayoutGuide](https://developer.apple.com/documentation/uikit/uikeyboardlayoutguide) @samwyndham thks for the suggestion